src/components/GoogleSignIn/GoogleSignIn.tsx

```diff
@@ -24,6 +24,9 @@ function GoogleSignIn(props: Props) {
         setGoogleId(response.tokenId);
         api.setGoogle(response.tokenId).then(() => {
             props.onAfterLogin();
+        }).catch(() => {
+            googleId = null;
+            localStorage.removeItem("googleId");
```

Comment:
And rerender the page

Reply:
The GoogleLogin-Component does rerender if the googleId is set to null.
If you refer to the parent-Component. I cant, dont want and dont need to rerender this.
The onAfterLogin-Function is never executed, so the parent knows, it hasnt logged in